tests/unit_tests/utils/test_general.py

```diff
@@ -41,6 +41,12 @@ def test_collect_dict_data(io_handler, caplog) -> None:
     assert isinstance(_dict, dict)
     assert len(_dict) > 0
 
+    # file with several documents
```

Comment:
Maybe you can add a test here to check that `yaml.composer.ComposerError` is thrown?

Reply:
We don't raise ` yaml.composer.ComposerError`. This error is thrown by yaml if you try to open a multi-document yaml file with `yaml.load`. In this case, we want to load all documents with `yaml.safe_load_all`. So the test using a multi-document yaml is covering this.